common4j/src/main/com/microsoft/identity/common/java/controllers/ExceptionAdapter.java

```diff
@@ -416,6 +423,40 @@ public static ClientException clientExceptionFromException(@NonNull final Throwa
             );
         }
 
+        if (e instanceof GeneralSecurityException) {
+            if (e instanceof CertificateException) {
+                return new ClientException(
+                        ClientException.CERTIFICATE_LOAD_FAILURE,
+                        e.getMessage(),
+                        e);
+            } else if (e instanceof KeyStoreException) {
+                return new ClientException(
+                        ClientException.KEYSTORE_NOT_INITIALIZED,
+                        e.getMessage(),
+                        e);
+            } else if (e instanceof NoSuchAlgorithmException) {
+                return new ClientException(
+                        ClientException.NO_SUCH_ALGORITHM,
+                        e.getMessage(),
+                        e);
+            } else if (e instanceof InvalidAlgorithmParameterException) {
+                return new ClientException(
+                        ClientException.INVALID_ALG_PARAMETER,
+                        e.getMessage(),
+                        e);
+            } else if (e instanceof UnrecoverableEntryException) {
+                return new ClientException(
+                        ClientException.INVALID_PROTECTION_PARAMS,
+                        e.getMessage(),
+                        e);
+            } else if (e instanceof InvalidKeyException) {
+                return new ClientException(
+                        ClientException.INVALID_KEY,
+                        e.getMessage(),
+                        e);
+            }
```

Comment:
I assume we can't move this to a switch statement. I think java 17 introduced a way to emulate instanceof in a switch statement, I'm not sure if that's supported in our default java version though.

Reply:
I think you're talking about this? https://docs.oracle.com/en/java/javase/17/language/pattern-matching-switch-expressions-and-statements.html
Looks like we would need to enable it explicitly as a preview (I did check to see if it would automatically work in our codebase). Would be cool if this made it to the next prod Java version.